### PR TITLE
feat: display coinmarket info notes as tags

### PR DESCRIPTION
**Pull request by yanascz**
Display coinmarket info notes as tags.

## Description

Coinmarket info notes starting with # are converted to tags, see screenshots below.

## Screenshots:

image
image

### CODE REVIEW

File: packages/suite/src/views/wallet/coinmarket/buy/offers/Offers/List/Quote/index.tsx [hunk 1]
```diff
@@ -212,12 +215,11 @@ const Quote = ({ className, quote, wantCrypto }: Props) => {
     const { selectQuote, providersInfo } = useCoinmarketBuyOffersContext();
     // TODO - tags are not yet fully supported by the API server
```
Comment (martinboehm) [inline packages/suite/src/views/wallet/coinmarket/buy/offers/Offers/List/Quote/index.tsx]: I would remove the TODO from here - we are handling the visual tag at this moment and when/if we extend the `getTagAndInfoNote`, we will know what to do.

Comment (yanascz) [reply packages/suite/src/views/wallet/coinmarket/buy/offers/Offers/List/Quote/index.tsx]: Removed.

**Comment by martinboehm** [comment]
Looks good to me

File: packages/suite/src/components/wallet/CoinmarketTag/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    text-transform: uppercase;
+`;
+
+interface Props {
+    tag?: string;
+}
+
+const CoinmarketTag = ({ tag }: Props) => <TagRow>{tag && <Tag>{tag}</Tag>}</TagRow>;
+
+export default CoinmarketTag;
```
Comment (matejkriz) [inline packages/suite/src/components/wallet/CoinmarketTag/index.tsx]: We want to stop using `export default` and `index.tsx` files, reasons explained [in internal Notion wiki ](https://www.notion.so/satoshilabs/Code-Improvements-1-Code-style-changes-0f1bb44d86764715b8f5962c7f23466e). 

Despite the temporary inconsistency – there is no other way to get rid of it.

Our usual approach is to create separate followup PR with just refactoring of the touched part. I understand that there's a lot of components here to refactor, so I would suggest to stop using it at least in this new component. What do you think?

Comment (matejkriz) [inline packages/suite/src/components/wallet/CoinmarketTag/index.tsx]: Props ⇒ ComponentProps 
one more nitpick - please start prefixing Props type by Component name. No big deal but it’s more explicit, easier to export type later, nicer error messages...

```suggestion
interface CoinmarketTagProps {
    tag?: string;
}

export const CoinmarketTag = ({ tag }: CoinmarketTagProps) => <TagRow>{tag && <Tag>{tag}</Tag>}</TagRow>;
```

File: packages/suite/src/utils/wallet/coinmarket/__tests__/coinmarketUtils.test.ts [hunk 3]
```diff
@@ -81,4 +82,12 @@ describe('coinmarket utils', () => {
             },
         ]);
     });
+
+    it('getTagAndInfoNote', () => {
+        expect(getTagAndInfoNote({})).toStrictEqual({ infoNote: undefined });
+        expect(getTagAndInfoNote({ infoNote: '' })).toStrictEqual({ infoNote: '' });
+        expect(getTagAndInfoNote({ infoNote: 'Foo' })).toStrictEqual({ infoNote: 'Foo' });
```
Comment (matejkriz) [inline packages/suite/src/utils/wallet/coinmarket/__tests__/coinmarketUtils.test.ts:89]: nit: consider adding one more test case to explicitly show that `#` matters only at the beginning.

Are the trailing spaces handled for infoNote somewhere? What about `infoNote: ' #Foo'`?

```suggestion
        expect(getTagAndInfoNote({ infoNote: 'Foo' })).toStrictEqual({ infoNote: 'Foo' });
        expect(getTagAndInfoNote({ infoNote: 'Foo#Bar' })).toStrictEqual({ infoNote: 'Foo#Bar' });
```

**Review by matejkriz — APPROVED** [review]
Only nitpicks/suggestions - feel free to close them if you don't find them relevant for now.

Is there any testing API with some `infoNote` already available? I quickly tried staging-exchange1 and 2, but see no tags...

**Comment by martinboehm** [comment]
@matejkriz Thank you for the review! I set up some test info notes in the staging-exchange2 environment (and during that fixed a tricky bug in the exchange info note display), if you want to check the functionality. The rest of the issues will be fixed by @yanascz next week , when he is back from vacation (I think there is still time to the freeze).

**Comment by matejkriz** [comment]
> @matejkriz Thank you for the review! I set up some test info notes in the staging-exchange2 environment (and during that fixed a tricky bug in the exchange info note display), if you want to check the functionality. The rest of the issues will be fixed by @yanascz next week , when he is back from vacation (I think there is still time to the freeze).

Thanks @martinboehm, looks good! As I mentioned, my comments are only nitpics/suggestions. PR is already approved so we don't need to worry about the freeze.

File: patches/@types+invity-api+1.0.6.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+diff --git a/node_modules/@types/invity-api/index.d.ts b/node_modules/@types/invity-api/index.d.ts
```
Comment (Nodonisko) [inline patches/@types+invity-api+1.0.6.patch:1]: Uff more patches, why not publish new version of your package with fixed types?

Comment (yanascz) [reply patches/@types+invity-api+1.0.6.patch:1]: @martinboehm, could you please answer this one?

Comment (martinboehm) [reply patches/@types+invity-api+1.0.6.patch:1]: @Nodonisko this is only a small fix in the types. We will release a new version of the types with the next bigger feature in about one month, the patch is only very temporary.

**Comment by yanascz** [comment]
@matejkriz, I addressed your comments. Please let me know I can squash the changes.

**Comment by matejkriz** [comment]
> @matejkriz, I addressed your comments. Please let me know I can squash the changes.

Cool, thanks! Squash&merge 🚀

**Pull request merged**